bpf/bpf_syscall.go

```diff
@@ -367,3 +376,46 @@ func DeleteMapEntry(mapFD MapFD, k []byte, valueSize int) error {
 	}
 	return nil
 }
+
+// GetMapNextKey returns the next key for the given key if the current key exists.
+// Otherwise it returns the first key. Order is implemention / map type
+// dependent.
+//
+// Start iterating by passing a nil key.
+func GetMapNextKey(mapFD MapFD, k []byte, keySize int) ([]byte, error) {
```

Comment:
Might be worth having a stateful iterator object for this so we can avoid allocating and copying for every iteration?

Reply:
I will defer this into another PR. We can do the same for lookup and we can make the `PinnedMap.Iter()` allocate the memory once and we may also use it for deletes etc. So might be a bigger standalone change.